plugins/ffxiv/ffxiv.cpp

```diff
@@ -0,0 +1,203 @@
+// Copyright 2005-2016 The Mumble Developers. All rights reserved.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file at the root of the
+// Mumble source tree or at <https://www.mumble.info/LICENSE>.
+
+#include "../mumble_plugin_win32_32bit.h" // Include standard plugin header.
+#include "../mumble_plugin_utils.h" // Include plugin header for special functions, like "escape".
+#include <cmath>
+
+// Offset values can be obtained from:
+//   http://xivapp.com/api/structures?patchVersion=latest&platform=x86
+// Y is North/South, South is increasing values
+// X is East/West, West is increasing values
+// Heading is in radians, 0 faces south
+
+// Memory offsets
+const procptr32_t camera_ptr              = 0x1045C40;
+const procptr32_t avatar_ptr              = 0x10468EC;
+const procptr32_t state_offset            = 0x1048C60;
+const procptr32_t map_id_offset           = 0x10210B0;
+// Avatar struct offsets
+const procptr32_t identity_offset         = 48;  // Name
+const procptr32_t avatar_pos_offset       = 160; // X, Z, Y
+const procptr32_t avatar_azimuth_offset   = 176; // Heading (-pi to pi)
+// Camera struct offsets
+const procptr32_t camera_is_free_offset   = 256; // 0: First person mode; 1: 3rd person
+const procptr32_t camera_pos_offset       = 64;  // X, Z, Y
+const procptr32_t camera_azimuth_offset   = 288; // (-pi to pi)
+const procptr32_t camera_elevation_offset = 292; // (-pi to pi)
+// Module names
+const wchar_t *exe_name                 = L"ffxiv.exe";
+
+static int fetch(float *avatar_pos, float *avatar_front, float *avatar_top, float *camera_pos, float *camera_front, float *camera_top, std::string &context, std::wstring &identity) {
+	for (int i=0;i<3;i++) {
+		avatar_pos[i] = avatar_front[i] = avatar_top[i] = camera_pos[i] = camera_front[i] = camera_top[i] = 0.0f;
+	}
+
+	// Boolean value to check if game addresses retrieval is successful
+	bool ok;
+	// Create containers to stuff our raw data into, so we can convert it to Mumble's coordinate system
+	float avatar_pos_corrector[3], camera_pos_corrector[3], avatar_azimuth, camera_azimuth, camera_elevation;
+	// Values for extra features
+	char player[64];
+	unsigned int map_id;
+	// State
+	unsigned char state, camera_is_free;
+
+	// Avatar pointer
+	procptr32_t avatar_address = peekProc<procptr32_t>(pModule + avatar_ptr);
+	if (!avatar_address) return false;
+
+	// Camera pointer
+	procptr32_t camera_address = peekProc<procptr32_t>(pModule + camera_ptr);
+	if (!camera_address) return false;
+
+	// Peekproc and assign game addresses to our containers, so we can retrieve positional data
+	ok = peekProc(pModule + state_offset, &state, 1) && // Magical state value: 0 or 255 when in main menu and 1 when in-game.
```

Comment:
Ok, I'm aligning all the `&& //` so I'll use tabs for the indentation, and spaces to align those.